templates/create-daml-app-test-resources/index.test.ts

```diff
@@ -0,0 +1,305 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { ChildProcess, spawn, SpawnOptions } from 'child_process';
+import waitOn from 'wait-on';
+
+import Ledger from '@daml/ledger';
+import { User } from '@daml.js/create-daml-app';
+import { computeCredentials } from './Credentials';
+
+import puppeteer, { Browser, Page } from 'puppeteer';
+
+const SANDBOX_PORT = 6865;
+const JSON_API_PORT = 7575;
+const UI_PORT = 3000;
+
+// `daml start` process (which spawns a sandbox and JSON API server)
+let startProc: ChildProcess | undefined = undefined;
+
+// Headless Chrome browser:
+// https://developers.google.com/web/updates/2017/04/headless-chrome
+let browser: Browser | undefined = undefined;
+
+let uiProc: ChildProcess | undefined = undefined;
+
+// Function to generate unique party names for us.
+// This should be replaced by the party management service once that is exposed
```

Comment:
Let’s not change this in this PR but the party management service is exposed now.